Plugins/GitUIPluginInterfaces/IGitUICommands.cs

```diff
@@ -156,5 +156,7 @@ public interface IGitUICommands
         bool StartVerifyDatabaseDialog();
         bool StartViewPatchDialog();
         bool StartSparseWorkingCopyDialog();
+        void AddCommitTemplate(string key, Func<string> addingText);
+        void RemoveCommitTemplate(string key);
```

Comment:
I don't believe these definitions or the implementing methods are needed - registered templates are stored in a static dictionary so they are accessible universally.

In the plugin you do:

```
private readonly ICommitTemplateManager  _commitTemplateManager = new CommitTemplateManager();

_commitTemplateManager.Register(...);
```

and in `FormCommit` you call: 

```
private readonly ICommitTemplateManager  _commitTemplateManager = new CommitTemplateManager();

var list = _commitTemplateManager.Registered.Select(x => new { Template = x, Name = x.Name })...
```